### PR TITLE
Optimize: move experimental cmd to alpha stage

### LINKED ISSUE #1522: Support sealer alpha command to incubate some alpha features
Hi, All,

sealer aims to solve distributed application packaging/distributing/running issues. Currently, it has already supported many command, like, run, apply, build and so on.

While, when receiving more and more feature requests, sealer community found that sealer could accept some of them, but the feature may need long-term design, experiment and improvement. In a word, some features may still be early stage.

For these feature request, sealer community takes quite open attitude, and wishes to open another policy to support this--`sealer alpha` command. For example, sealer maintainer discussed the `take-over` command should be included in `sealer alpha`, since it is not so mature, but still wish more end-users could experience and input more advices.

Here are some features that we plan to include in alpha command.

- [ ] take-over command
- [ ] upgrade command
- [ ] no-underlying-base-support

Welcome all participants to input more ideas about alpha command

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it

move below experimental cmd to alpha stage to incubate.

1. `sealer prune`
2. `sealer gen`
3. `sealer debug`
4. `sealer exec`
5. `sealer cert`
6. `sealer merge`
7. `sealer upgrade`

### Does this pull request fix one issue?

fix #1522

### CODE REVIEW

File: cmd/sealer/cmd/alpha/alpha.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import (
+	"github.com/spf13/cobra"
+)
+
+// NewCmdAlpha returns "sealer alpha" command.
+func NewCmdAlpha() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "alpha",
+		Short: "Sealer experimental sub-commands",
```
Comment (allencloud) [inline cmd/sealer/cmd/alpha/alpha.go:25]: Please add a long field to add more description of `sealer alpha` command. This is very useful for end users. Thanks. And do not make sealer uppercase. sealer is a reserved word, then it has no uppercase mode, just like etcd. sealer and etcd both have a lowercase first letter.

File: cmd/sealer/cmd/alpha/cert.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	"github.com/spf13/cobra"
+
+	"github.com/sealerio/sealer/common"
+	"github.com/sealerio/sealer/pkg/clusterfile"
+	"github.com/sealerio/sealer/pkg/runtime"
+)
+
+var altNames []string
+
+var longCertCmdDescription = `Add domain or ip in certs: you had better backup old certs first. this command will update cluster API server cert, you need to restart your API server manually after using sealer cert. then, you can using cmd "openssl x509 -noout -text -in apiserver.crt" to check the cert details.
```
Comment (allencloud) [inline cmd/sealer/cmd/alpha/cert.go:29]: Please make sure that first word of every sentence should be uppercase. It is minor mistake. But I think we should pay more attention to this minor cases.

File: cmd/sealer/cmd/alpha/cert.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+func NewCertCmd() *cobra.Command {
+	certCmd := &cobra.Command{
+		Use:     "cert",
+		Short:   "Update Kubernetes API server's cert",
+		Args:    cobra.NoArgs,
+		Long:    longCertCmdDescription,
+		Example: exampleForCertCmd,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(altNames) == 0 {
+				return fmt.Errorf("ip address or DNS domain needed for cert Subject Alternative Names")
```
Comment (allencloud) [inline cmd/sealer/cmd/alpha/cert.go:48]: s/ip/IP/g, IP is a reserved word. While I think quite lot of people use ip, which is lowercase. And lowercase is incorrect.

File: cmd/sealer/cmd/alpha/cert.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+			r, err := runtime.NewDefaultRuntime(cluster, clusterFile.GetKubeadmConfig())
+			if err != nil {
+				return fmt.Errorf("failed to get default runtime: %v", err)
+			}
+
+			return r.UpdateCert(altNames)
+		},
+	}
+
+	certCmd.Flags().StringSliceVar(&altNames, "alt-names", []string{}, "add domain or ip in certs, sealer.cool or 10.103.97.2")
```
Comment (allencloud) [inline cmd/sealer/cmd/alpha/cert.go:70]: Do we have a default domain name? If we have, we should add it explicitly in the flag description.

File: cmd/sealer/cmd/alpha/exec.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+	}
+
+	if len(roles) == 0 {
+		ipList = append(cluster.GetMasterIPList(), cluster.GetNodeIPList()...)
+	} else {
+		for _, role := range roles {
+			ipList = append(ipList, cluster.GetIPSByRole(role)...)
+		}
+		if len(ipList) == 0 {
+			return fmt.Errorf("failed to get target ipList by roles label %s", roles)
```
Comment (allencloud) [inline cmd/sealer/cmd/alpha/exec.go:79]: should add the detailed failure reason:

```
return fmt.Errorf("failed to get target ipList: no IP gotten by role(%s)", roles)
```

**Comment by allencloud** [comment]
`cmd/sealer/cmd/alpha/merge.go:38: File is not `gofmt`-ed with `-s` (gofmt)` @kakaZhou719

File: cmd/sealer/cmd/alpha/alpha.go (hunk 1 above)
Comment (kakaZhou719) [reply cmd/sealer/cmd/alpha/alpha.go:25]: `alpha` command has sub command which has their own long description. i am ok with adding long description on `alpha` command, but seems like there is very little to describe.

File: cmd/sealer/cmd/alpha/cert.go (hunk 4 above)
Comment (kakaZhou719) [reply cmd/sealer/cmd/alpha/cert.go:70]: yes, we have a default domain in the cert process builtin, i will add it explicitly in the flag description.

File: cmd/sealer/cmd/alpha/alpha.go (hunk 1 above)
Comment (allencloud) [reply cmd/sealer/cmd/alpha/alpha.go]: 😄， we should explain to help end-users to understand what is alpha command for. Please take the following paragraph into consideration.

```
alpha command of sealer is used to provide functionality incubation from immature to mature. Each function will experience a growing procedure. Alpha command policy calls on end users to experience alpha functionality as early as possible, and actively feedback the experience results to sealer community, and finally cooperate to promote function from incubation to graduation. Please file an issue at https://github.com/sealerio/sealer/issues when you have any feedback on alpha commands.
```

File: cmd/sealer/cmd/alpha/cert.go [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+			r, err := runtime.NewDefaultRuntime(cluster, clusterFile.GetKubeadmConfig())
+			if err != nil {
+				return fmt.Errorf("failed to get default runtime: %v", err)
+			}
+
+			return r.UpdateCert(altNames)
+		},
+	}
+
+	certCmd.Flags().StringSliceVar(&altNames, "alt-names", []string{}, "add DNS domain or IP in certs, if it already in the cert subject alternative names list, nothing will changed")
```
Comment (allencloud) [inline cmd/sealer/cmd/alpha/cert.go:74]: add DNS domain or IP in certs. If it is already in the cert subject alternative names list, nothing will be changed.

File: cmd/sealer/cmd/alpha/prune.go [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/sealerio/sealer/pkg/prune"
+
+	"github.com/spf13/cobra"
+)
+
+var exampleForPruneCmd = `The following command will prune sealer data directory such as, image db ,image layer, build tmp.
```
Comment (allencloud) [inline cmd/sealer/cmd/alpha/prune.go:27]: `The following command will prune sealer data directory, such as: image db, image layer and build tmp.`

**Review by allencloud — APPROVED** [review]

**Pull request merged**